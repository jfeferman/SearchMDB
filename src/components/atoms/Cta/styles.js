import styled from 'styled-components';
import { px2rem, colors } from '../../styleUtils/globalStyleVariables';

export const StyledButton = styled.button`
  border-radius: ${px2rem(10)};
  border: 1px solid;
  cursor: pointer;
  display: block;
  font-weight: 300;
  text-align: center;
  text-decoration: none;
  white-space: nowrap;
  width: auto;
  background: ${colors.lightGray};
  color: ${colors.black};
  font-size: ${px2rem(15)};
  line-height: ${px2rem(17)};
  letter-spacing: ${px2rem(0.34)};
  padding: ${px2rem(16)} ${px2rem(41.5)};
  vertical-align: middle;
  outline: 0;
  transition: color .15s ease-in-out,
    background-color .15s ease-in-out,
    border-color .15s ease-in-out;

  &:focus {
    box-shadow: 0px 0px 2px 2px rgba(23,109,220,0.8);
  }

  ${(props) => props.disabled && `
    cursor: not-allowed;
    background: ${colors.disabled};
    color: ${colors.disabledText};
    border: 0;
    &:hover {
      color: ${colors.disabledText};
    }
  `};
`;
