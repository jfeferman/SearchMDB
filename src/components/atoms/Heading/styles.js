import styled from 'styled-components';
import { fonts } from '../../styleUtils/globalStyleVariables';

export const StyledH1 = styled.h1`
  ${fonts.H1}
`;

export const StyledH2 = styled.h2`
  ${fonts.H2}
`;

export const StyledH3 = styled.h3`
  ${fonts.H3}
`;

export const StyledH4 = styled.h4`
  ${fonts.H4}
`;
