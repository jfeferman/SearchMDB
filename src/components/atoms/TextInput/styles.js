import styled from 'styled-components';
import { px2rem, colors } from '../../styleUtils/globalStyleVariables';

export const StyledTextInput = styled.input`
  width: 100%;
  box-sizing: border-box;
  height: ${px2rem(56)};
  border: 1px solid ${colors.lightGray};
  border-radius: ${px2rem(3)};
  background-color: ${colors.white};
  padding-left: ${px2rem(20)};
  font-size: ${px2rem(16)};
`;
