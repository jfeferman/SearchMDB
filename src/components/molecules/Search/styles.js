import styled from 'styled-components';
import { px2rem } from '../../styleUtils/globalStyleVariables';

export const StyledSearch = styled.div`
  display: flex;
  flex-direction: row;
  margin: 0 auto;
  width: ${px2rem(600)};

  div:nth-of-type(1) {
    margin-right: ${px2rem(10)};
  }
`;
