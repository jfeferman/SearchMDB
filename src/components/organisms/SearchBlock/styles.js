import styled from 'styled-components';
import { px2rem } from '../../styleUtils/globalStyleVariables';

export const StyledSearchBlock = styled.div`
  margin: ${px2rem(100)} auto ${px2rem(100)} auto;
  
  h1 {
    text-align: center;
  }
`;
