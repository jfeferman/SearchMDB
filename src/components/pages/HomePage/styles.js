import styled from 'styled-components';
import { px2rem } from '../../styleUtils/globalStyleVariables';

export const StyledResultsBlock = styled.div`
  margin: 0 auto;
  max-width: ${px2rem(980)};
  padding: 0 ${px2rem(40)} ${px2rem(80)} ${px2rem(40)};

  h3 {
    margin-bottom: ${px2rem(4)};
  }
`;
