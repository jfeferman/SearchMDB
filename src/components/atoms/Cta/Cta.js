import React from 'react';
import PropTypes from 'prop-types';
import { StyledButton } from './styles';

/**
 * Call to action button
 *
 * @property {boolan} disabled
 * @property {node} children
 */
const Cta = (props) => {
  const {
    children,
    disabled,
    onClick,
  } = props;

  return (
    <StyledButton
      type="button"
      disabled={disabled}
      onClick={onClick}
    >
      {children}
    </StyledButton>
  );
};

Cta.propTypes = {
  disabled: PropTypes.bool,
  children: PropTypes.node,
  onClick: PropTypes.func,
};

Cta.defaultProps = {
  disabled: false,
  children: [],
};

export default Cta;
