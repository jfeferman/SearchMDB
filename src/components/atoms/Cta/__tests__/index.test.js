import React from 'react';
import 'jest-styled-components';
import { create, act } from 'react-test-renderer';
import Cta from '..';
import { StyledButton } from '../styles';

describe('components/atoms/Cta', () => {
  let renderer;
  let instance;
  let input;
  let disabled = false;
  const text = 'Test Button';
  const onClick = jest.fn();

  beforeEach(() => {
    renderer = create(
      <Cta disabled={disabled} onClick={onClick}>
        {text}
      </Cta>,
    );
    instance = renderer.root;
    input = instance.find(el => el.type === StyledButton);
  });

  it('matches the snapshot of the component', () => {
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('verifies that a styled button component is present', () => {
    expect(instance.findAll(el => el.type === StyledButton)).toHaveLength(1);
  });

  it('verifies that the button text matches', () => {
    expect(instance.props.children).toBe(text);
  });

  it('verifies that the button is enabled', () => {
    expect(instance.props.disabled).toBe(false);
  });

  it('verifies function is called when button is clicked', () => {
    act(() => {
      input.props.onClick();
    });
    expect(onClick).toHaveBeenCalled();
  });

  it('updates the input to disabled and verifies disabled prop', () => {
    disabled = true;
    renderer.update(
      <Cta disabled={disabled} onClick={onClick}>
        {text}
      </Cta>,
    );
    expect(instance.props.disabled).toBe(true);
  });
});
