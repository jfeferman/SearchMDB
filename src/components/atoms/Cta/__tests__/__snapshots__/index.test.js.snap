// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atoms/Cta matches the snapshot of the component 1`] = `
.c0 {
  border-radius: 6.25rem;
  border: 1px solid;
  cursor: pointer;
  display: block;
  font-weight: 300;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  width: auto;
  color: #ffffff;
  font-size: 0.9375rem;
  line-height: 1.0625rem;
  -webkit-letter-spacing: 0.02125rem;
  -moz-letter-spacing: 0.02125rem;
  -ms-letter-spacing: 0.02125rem;
  letter-spacing: 0.02125rem;
  padding: 1rem 2.59375rem;
  vertical-align: middle;
  outline: 0;
  -webkit-transition: color .15s ease-in-out, background-color .15s ease-in-out, border-color .15s ease-in-out;
  transition: color .15s ease-in-out, background-color .15s ease-in-out, border-color .15s ease-in-out;
}

.c0:focus {
  box-shadow: 0px 0px 2px 2px rgba(23,109,220,0.8);
}

<button
  className="c0"
  disabled={false}
  onClick={[MockFunction]}
  type="button"
>
  Test Button
</button>
`;
